### ISSUE TITLE
Add support for lesspass as password generation method 

### ISSUE THREAD

**Issue by bumi** [gopasspw/gopass#599]
[Lesspass](https://github.com/lesspass/lesspass) (or [similar](http://masterpasswordapp.com/algorithm.html) ideas) use a hashing function to generate a password from the site name, account name, nonce, etc. and a master password. (see [how lesspass works](https://blog.lesspass.com/lesspass-how-it-works-dde742dd18a4) )

What do you think about offering such a method as password generation algorithm? 

#### Something like: 

```
password = hash-function(name, account, nonce, master pass)
```` 

#### possible usage: 

```
gopass genrate foo/bar --calculate
service name:  google
account name: foobar@googlemail.com
nonce [1]:  2
secret: topsecret
````

**Comment by dominikschulz** [comment]
Right now gopass is fully focused on the combination of gpg and git. But we could think about adding different backends, e.g. one based on [Argon2](https://godoc.org/golang.org/x/crypto/argon2) that generates passwords from a master hash. I'm not really a fan of this approach, but I don't have any issues with it either.

The most important aspect of that is IMHO useability, because bad useability increases the change for mistakes (from users or developers). So my idea would be to only support one backend per mount. That way we could keep mostly pass compatible mounts as they are right now and add new mounts / sub-stores for different backends (e.g. KDF or database (Vault) based ones).

**Comment by bumi** [comment]
yeah, I think it should still be focused on that. 
I was thinking of some option like the `--xkcd` option. A certain password would be generated that then is stored as any other generated password right now. Imo it would not change anything on the security model of gopass. 

I would add another option in: https://github.com/justwatchcom/gopass/blob/f6b0ea19585dadf25b06c4c228aa558d08ca8c23/action/generate.go#L109

**Comment by falschparker82** [comment]
I also don't see the clear upside here for both gopass and lesspass users here.

Any generation based password managers have the same, serious drawbacks:
- No or Limited password rotation capabilities
- No or Limited special requirements capabilities (length, special chars, etc.)
(both of the above are partially covered by implementations, but you need to keep both the options and the password generation for every site in your head)
- Master password breach will leak all passwords, also for the future (especially bad for the gopass team case)

The one big upside is that (theoretically) they are very lightweight and cross-platform, as you only need to memorize the master password.

So it's a lose-lose situation here: If you've arrived at the point of needing a full-blown password manager already (with all the trouble in syncing secrets cross devices and managing certificates), there's no upside to go for the lesspass based generation. Quite the opposite, a fully random password will have better protections against all kinds of extra holes the generation based approach might open up (predicting future passwords etc.).

In other words, both concepts have a very different profile of strenghts and weaknesses (and are in a way mutually incompatible) — if you try to shove both together, you'll get none of the strengths of lesspass, but all the weaknesses.

Maybe I need a clearer picture of what you're trying to accomplish or make better exactly?

**Comment by dominikschulz** [comment]
If you want another password generation method, that would certainly possible, but why? I don't see the benefit, either.

**Comment by bumi** [comment]
thanks for the detailed input and information. I agree about the drawbacks of a password generation method. But actually in the gopass context I'd see this option only as a different password generation method - similar to the `--xkcd` option. Like now it is up to the user which password is stored. 

I am using some of the generated passwords to have a way to recover some passwords in the case I am loosing access to gopass. I like the generation of the password by some different variable input - getting a unique password (with symbols etc) from some memorable input (but not using the memorable input directly)

right now I created the passwords manually and saved them manually to gopass.

**Comment by dominikschulz** [comment]
I've tried a few possible implementations for this but I don't think the lesspass approach fits gopass very well. We've already discussed the algorithm but furthermore I don't think that it can be integrated nicely into gopass.

If you want you can still work on a PR otherwise we'll close this issue some time.
